Sources/Apodini/Semantic Model Builder/GraphQL/GraphQLInterfaceExporter.swift

```diff
@@ -0,0 +1,150 @@
+                                              const graphQLFetcher = graphQLParams =>
+                                                fetch('/graphql', {
+                                                  method: 'post',
+                                                  headers: { 'Content-Type': 'application/json' },
+                                                  body: JSON.stringify(graphQLParams),
+                                                })
+                                                  .then(response => response.json())
+                                                  .catch(() => response.text());
+                                              ReactDOM.render(
+                                                React.createElement(GraphiQL, { fetcher: graphQLFetcher }),
+                                                document.getElementById('graphiql'),
+                                              );
+                                            </script>
+                                          </body>
+                                        </html>
+                                        """
+
+        return Vapor.Response(status: .ok, headers: ["Content-Type": "text/html"], body: html)
+    }
+
+    private func graphqlServer(_ req: Vapor.Request) throws -> EventLoopFuture<String> {
+        guard let body = req.body.string else {
+            throw NetworkError.noBody
+        }
+
+        // Create Swift dict
+        var query = "{}"
+        let data = body.data(using: .utf8)!
+        do {
+            if let jsonArray = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: AnyObject] {
+                if let q = jsonArray["query"] as? String {
+                    query = q
+                } else {
+                    print("jsonArray doesn't have query parameter")
+                }
+
+            } else {
+                print("bad json")
+                throw NetworkError.noBody
+            }
+        } catch let error as NSError {
+            print(error)
+            throw NetworkError.noBody
+        }
+
+        return try graphql(schema: self.schema,
+                request: query,
+                context: req,
+                eventLoopGroup: req.eventLoop).map { result -> String in
+            result.description
+        }
+
+    }
+
+    // We should have parameter for the result struct values and paramters
+    func export<H: Handler>(_ endpoint: Endpoint<H>) {
+        self.graphQLPath.append(for: endpoint, with: endpoint.createConnectionContext(for: self))
+    }
+
+    func exportParameter<Type: Codable>(_ parameter: EndpointParameter<Type>) -> String {
+        parameter.name
+    }
+
+    func finishedExporting(_ webService: WebServiceModel) {
+        self.schema = self.graphQLPath.generate()
+    }
+
+    func retrieveParameter<Type: Decodable>(_ parameter: EndpointParameter<Type>, for request: GraphQLRequest) throws -> Type?? {
```

Comment:
Why double optional? Does `.none` here mean something else than `.some(.none)`

Reply:
The double optional is a requirement of the Exporter API.
See https://github.com/Apodini/Apodini/pull/81#discussion_r548919891.
We decided after a long call on this in order to distinguish between "value does not exist" (`nil`) and "nil is the value" `Optional.some(nil)` (for there later there exist a `.null` shortcut).